proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java

```diff
@@ -330,18 +330,38 @@ private void loadPlugins() {
 
     try {
       Path pluginPath = Path.of("plugins");
+      Path updatePath = pluginPath.resolve("to-update");
+      Path outdatedPluginsPath = pluginPath.resolve("outdated");
 
       if (!pluginPath.toFile().exists()) {
         Files.createDirectory(pluginPath);
-      } else {
-        if (!pluginPath.toFile().isDirectory()) {
-          logger.warn("Plugin location {} is not a directory, continuing without loading plugins",
-              pluginPath);
-          return;
-        }
+      }
+      if (!updatePath.toFile().exists()) {
+        Files.createDirectory(updatePath);
+      }
+      if (!outdatedPluginsPath.toFile().exists()) {
+        Files.createDirectory(outdatedPluginsPath);
+      }
+      if (!pluginPath.toFile().isDirectory()) {
```

Comment:
Instead of using `toFile()` everywhere, I think it would be better to use methods of the [Files](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/Files.html) class since the type of object used is Path. You can replace `File#isDirectory` with [Files#isDirectory(Path)](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/Files.html#isDirectory(java.nio.file.Path,java.nio.file.LinkOption...)) and `File#exists` with [Files#exists(Path)](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/Files.html#exists(java.nio.file.Path,java.nio.file.LinkOption...))